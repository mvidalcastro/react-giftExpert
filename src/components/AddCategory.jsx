import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategorias}) => {

    const [inputValue, setInputValue] = useState('Ingrese la búsqueda acá ...')
    
    /// manejadores de Eventos
    const handleInputChange = (evento) => {
       // console.log(evento.target.value);
         setInputValue(evento.target.value);   
    }
    const handleSubmit =(evento) => {

        evento.preventDefault();
        if(inputValue.trim().length > 2) //Si el texto digitado es mayor a 2 caracteres No lo agrega a la lista
        {
            setCategorias( cat => {
                return [inputValue, ...cat]
            })
            setInputValue('');

        }

    }

  return (
    <form onSubmit={ (event) => handleSubmit(event) }>
        <input
            type="text"
            value={ inputValue } 
            onChange= { (event) => {handleInputChange(event)}}/>
    </form>
  )
}

AddCategory.propTypes = {

    setCategorias: PropTypes.func.isRequired
}

//export default AddCategory