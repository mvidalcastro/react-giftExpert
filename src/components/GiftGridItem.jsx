import React from 'react'
import PropTypes from 'prop-types'

//export const GiftGridItem = (imagen) => {
  export const GiftGridItem = ({id, title, url}) => {
   // const {id, title, url} = imagen;
  
  //console.log(id, title, url);
  return (
    <div className='card'>
        <p> { title } </p> 
        <img src= {url} alt = {title} />    
    </div>
  )
}

GiftGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired    
}

GiftGridItem.propTypesexprt  = {}

//export default GiftGridItem