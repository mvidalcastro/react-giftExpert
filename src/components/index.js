export * from './AddCategory';
export * from './GiftGridItem'
export * from './GisftGrid'