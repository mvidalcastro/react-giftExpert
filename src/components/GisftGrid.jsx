import PropTypes from 'prop-types'
import {GiftGridItem} from './GiftGridItem';
import { useFetchGifts } from '../hooks/useFetchGifts';

export const GisftGrid = ( {categoria} ) => {



    const { imagen, isLoading} = useFetchGifts ( categoria);

  return (
    <>
      <h3> { categoria } </h3>
      {
          isLoading && (<h2> Cargando...</h2>)
          
      }
      <div className='card-grid'>
      
          { imagen.map( ( img ) =>{ 
              return  <GiftGridItem 
                          key = {img.id}
                          { ...img }/> 
              })}   

      </div>
    </>
  )
}

GisftGrid.propTypes = {}

//export default GisftGrid