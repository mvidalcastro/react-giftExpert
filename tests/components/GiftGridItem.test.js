import { GiftGridItem } from "../../src/components/GiftGridItem"
//import { createRoot } from 'react-dom/client';
import {render, screen} from '@testing-library/react';

describe('<GiftGridItem/>: Pruebas de Componente ', ()=> {

  const titulo = "Esto es una prueba";
  const url_prueba = "http://www.pruebita.com/imagen.jpg";

      test('Debe hacer match con el snapshot', () => { 
        const { container } = render(<GiftGridItem title= { titulo } url= { url_prueba } />);      
        expect( container ).toMatchSnapshot();
      
      });


     test('Debe mostrar la imagen con la URL yel ALT indicado', () => { 
      render(<GiftGridItem title= { titulo } url= { url_prueba } />);   
     
     // expect(screen.getByRole('img').src).toBe(url_prueba);
     // expect(screen.getByRole('img').alt).toBe(titulo);
      const { src, alt } = screen.getByRole('img');
       expect(src).toBe(url_prueba);
       expect(alt).toBe(titulo);  
      })

      test('Debe mostrar el tÃ­tulo del componente', () => { 

        render(<GiftGridItem title= { titulo } url= { url_prueba } />); 
        expect(screen.getByText( titulo )).toBeTruthy();

       })

})