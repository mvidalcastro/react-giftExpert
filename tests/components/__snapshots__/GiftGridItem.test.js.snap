// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftGridItem/>: Pruebas de Componente  Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
       
      Esto es una prueba
       
    </p>
    <img
      alt="Esto es una prueba"
      src="http://www.pruebita.com/imagen.jpg"
    />
  </div>
</div>
`;
